### ISSUE TITLE
CP for LLMs 

### ISSUE THREAD

**Issue by pat-alt** [JuliaTrustworthyAI/ConformalPrediction.jl#65]
**Issue set up for Experiment Week** 

- [x] Study this [paper](https://arxiv.org/abs/2305.18404)
- [x] Get data from hugging face
- [x] ~Train small transformer model from scratch~
- [x] ~Look at fine-tuning of pre-trained model~

**Comment by pat-alt** [comment]
cc @florisdenhengst

**Comment by pat-alt** [comment]
*Notes from paper*

- **Context**: LLMs for the task of multiple-choice question answering (MCQA)
- Uncertainty, as estimated through CP, highly correlated with model accuracy
- Working with inductive/split conformal prediction to avoid retraining
- **Model**: LLaMA-13B
- **Data**: MMLU benchmark containing MCQA question from 57 domains, 4 possible answers each
- Find that set size is higher on average (closer to 4 for more difficult domains)
- Find that set size negatively correlates with top-1 accuracy and argue that this can be used to filter low-quality predictions
- Also find fairly robust size-stratified coverage, even though it does not appear they used adaptive prediction sets
- Show that coverage guarantees do not hold if exchangeability assumption is violated (see below)

![image](https://github.com/JuliaTrustworthyAI/ConformalPrediction.jl/assets/55311242/1f0d919b-c09a-40a7-b757-f029c312b919)

- Code released as notebook: https://github.com/bhaweshiitk/ConformalLLM/blob/main/Conformal_Prediction_for_LLMs.ipynb

**Comment by pat-alt** [comment]
For work in Julia, we probably want to look at [`Transformers.jl`](https://chengchingwen.github.io/Transformers.jl/dev/)
